apps/web/app/(app)/onboarding/components/onboarding.tsx

```diff
@@ -0,0 +1,168 @@
+
+      // Setup the event listener for the custom event to listen to survey completion
+      window.addEventListener("SurveyCompleted", handleSurveyCompletion);
+
+      // Cleanup function to remove the event listener
+      return () => {
+        window.removeEventListener("SurveyCompleted", handleSurveyCompletion);
+      };
+    }
+  }, [iframeVisible, CURRENT_STEP]); // Depend on iframeVisible and CURRENT_STEP to re-evaluate when needed
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      // Access localStorage only when window is available
+      const pathwayValueFromLocalStorage = localStorage.getItem("pathway");
+      const currentStepValueFromLocalStorage = parseInt(localStorage.getItem("CURRENT_STEP") ?? "1");
+
+      setSelectedPathway(pathwayValueFromLocalStorage);
+      SET_CURRENT_STEP(currentStepValueFromLocalStorage);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (CURRENT_STEP) {
+      const stepProgressMap = { 1: 16, 2: 50, 3: 65, 4: 75, 5: 90 };
+      const newProgress = stepProgressMap[CURRENT_STEP] || 16;
+      setProgress(newProgress);
+      localStorage.setItem("CURRENT_STEP", CURRENT_STEP.toString());
+    }
+  }, [CURRENT_STEP]);
+
+  // Function to render current onboarding step
+  const renderOnboardingStep = () => {
+    switch (CURRENT_STEP) {
+      case 1:
+        return (
+          <PathwaySelect
+            setSelectedPathway={setSelectedPathway}
+            SET_CURRENT_STEP={SET_CURRENT_STEP}
+            isFormbricksCloud={isFormbricksCloud}
+          />
+        );
+      case 2:
+        return (
+          selectedPathway !== "link" && (
+            <Role
+              setFormbricksResponseId={setFormbricksResponseId}
+              session={session}
+              SET_CURRENT_STEP={SET_CURRENT_STEP}
+            />
+          )
+        );
+      case 3:
+        return (
+          <Objective
+            formbricksResponseId={formbricksResponseId}
+            user={user}
+            SET_CURRENT_STEP={SET_CURRENT_STEP}
+          />
+        );
+      case 4:
+        return (
+          <ConnectWithFormbricks
+            environment={environment}
+            webAppUrl={webAppUrl}
+            SET_CURRENT_STEP={SET_CURRENT_STEP}
+            isFormbricksCloud={isFormbricksCloud}
+          />
+        );
+      case 5:
+        return selectedPathway === "link" ? (
+          <CreateFirstSurvey environmentId={environment.id} />
+        ) : (
+          <InviteTeamMate environmentId={environment.id} team={team} SET_CURRENT_STEP={SET_CURRENT_STEP} />
+        );
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div className="flex h-full w-full flex-col items-center bg-slate-50">
+      <OnboardingHeader progress={progress} />
+      <div className="mt-20 flex w-full justify-center bg-slate-50">
+        {renderOnboardingStep()}
+        {iframeVisible && (
+          <iframe
+            src="http://localhost:3000/s/clsui9a7x0000fbh5orp0g7c5"
```

Comment:
@mattinannt you need to create a survey with an animation or image bg locally and replace this URL to test the Link Survey onboarding path.

Before merging into prod, you need to add the link to the Onboarding Survey we have in the Formbricks Cloud account.

Now that I think about it, we should make use of the user identification in link surveys via URL to attribute the responses to the user who just signed up. We have the user email in the session object so it shouldn't be an issue. Let me try that first!